plugin-server/src/worker/ingestion/event-pipeline/1-populateTeamDataStep.ts

```diff
@@ -0,0 +1,35 @@
+import { PluginEvent } from '@posthog/plugin-scaffold'
+
+import { PipelineEvent } from '../../../types'
+import { EventPipelineRunner, StepResult } from './runner'
+
+/* 
+This step populates event.team_id and deletes event.ip if needed.
+If the event already has a team_id we will not run this step and 
+the capture endpoint will have handled this process. This is 
+temporary as this step will become the default for all events
+when we fully remove this functionality from the capture endpoint.
+*/
+export async function populateTeamDataStep(runner: EventPipelineRunner, event: PipelineEvent): Promise<StepResult> {
+    if (!event.token) {
+        runner.hub.statsd?.increment('dropped_event_with_no_team', { token_set: 'false' })
+        return null
+    }
+
+    const team = await runner.hub.teamManager.getTeamByToken(event.token)
+
+    // should we actually throw here?
+    if (!team) {
+        runner.hub.statsd?.increment('dropped_event_with_no_team', { token_set: 'true' })
```

Comment:
Would it be worth adding token there too?

Reply:
unsure - we probably expect few tokens to end up here but kind of goes against best practice for Grafana performance and usefulness